Tests/LibThreading/TestThread.cpp

```diff
@@ -29,10 +28,32 @@ TEST_CASE(threads_can_detach)
 
 TEST_CASE(joining_detached_thread_errors)
 {
-    auto thread = Threading::Thread::construct([]() { return 0; });
+    Atomic<bool> should_exit { false };
+    auto thread = Threading::Thread::construct([&]() {
+        while (!should_exit.load())
+            usleep(10 * 1000);
+        return 0;
+    });
     thread->start();
     thread->detach();
 
-    EXPECT(thread->join().is_error());
+    // Because of how the crash test forks and removes the thread, we can't use that to verify that join() crashes. Instead, we check the join crash condition ourselves.
+    EXPECT(!thread->needs_to_be_joined());
+
+    // FIXME: Dropping a running thread crashes because of the Function destructor. For now, force the detached thread to exit.
+    should_exit.store(true);
+    usleep(20 * 1000);
+}
+
+TEST_CASE(join_dead_thread)
+{
+    auto thread = Threading::Thread::construct([&]() { return 0; });
+    thread->start();
+    // The thread should have exited by then.
+    usleep(40 * 1000);
+
+    auto join_result = thread->join<int*>();
```

Comment:
this being int* when you returned an int is pretty sus. Should join use reinterpret_cast instead of static_cast?

Reply:
I think this is just to discard the value, as the 0 is for nullptr afaik. I will have a look next week once I have time.